### PR TITLE
Fixed battery bug in LTS (#203)

### LINKED ISSUE #203: LTS battery glitch
A very annoying LTS battery glitch which ruins the game mode, to clarify, the glitch occurs when the round ends with your pilot holding a battery, after the round ends, throughout the entirety of the match you'll be unable to pick a battery, noticeable by your battery icon appearing on your hud and in your titan.

### PR DESCRIPTION
**Pull request by ASpoonPlaysGames**
Remove all batteries from players at the end of a round for round-based modes, prevents client thinking that they both have and don't have a battery, resulting in weird stuff (ui disagrees with player model, etc.)

From my testing this does fix #203 however if modders want to make batteries persist through rounds it'll be a bit more of a pain now

### CODE REVIEW

**Comment by Dinorush** [comment]
Could you also chuck `PROTO_CleanupTrackedProjectiles( player )` into that loop? (player need not be alive)
Don't know if an issue was made about it, but Tethers persist through LTS rounds; that fixes it. Shouldn't be called for NPCs - in vanilla, auto-titan tethers are not cleared between rounds.

**Comment by ASpoonPlaysGames** [comment]
Done that, haven't tested just yet though

**Comment by Dinorush** [comment]
I tested it myself, it'll definitely clean up tethers. Not sure if there's other side effects, but I doubt it would be anything that doesn't occur in vanilla.

**Comment by ASpoonPlaysGames** [comment]
> in vanilla, auto-titan tethers are not cleared between rounds.
Respawn moment?

**Comment by Dinorush** [comment]
In all honesty the more I think about it, it probably _does_ get called for NPCs. But if the auto-titan dies, then its tethers obviously won't be cleared by the end of round cleanup, hence Tethers persisting for them.

**Comment by ASpoonPlaysGames** [comment]
So should i copy that line into the NPC loop as well?

**Comment by Dinorush** [comment]
Up to you. I clearly haven't tested whether NPC tethers get cleared in vanilla if they survive the round, but I really don't think it'll matter much either way considering I don't know for sure with how much I've played LTS.

**Comment by ASpoonPlaysGames** [comment]
I feel like even if they don't persist like that in vanilla, they probably *should*, so I'm gonna do it

**Comment by Dinorush** [comment]
Oh, I just looked up the method chain a bit more. Turns out those tracked projectiles only get tracked for players, so it definitely doesn't get cleared for NPCs.

**Comment by ASpoonPlaysGames** [comment]
Oh ok, will revert commit

File: Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut [hunk 1]
```diff
@@ -666,7 +666,8 @@ void function CleanUpEntitiesForRoundEnd()
 	foreach ( entity player in GetPlayerArray() )
 	{
 		ClearTitanAvailable( player )
-		
+		PROTO_CleanupTrackedProjectiles( player )
+		player.SetPlayerNetInt( "batteryCount", 0 )
```
Comment (GeckoEidechse) [inline Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut:670]:
```suggestion
		player.SetPlayerNetInt( "batteryCount", 0 ) 
```

The `PROTO_CleanupTrackedProjectiles`  should be it's own PR. The rest works and fixes our current problem with batteries ^^

**Review by GeckoEidechse — CHANGES_REQUESTED** [review]

File: Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut (hunk 1 above)
Comment (ASpoonPlaysGames) [reply Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut:670]: Ok, will revert

Comment (ASpoonPlaysGames) [reply Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut]: Done now
